src/components/user-activity.ts

```diff
@@ -116,14 +120,17 @@ export class UserActivityTracker {
         }
 
         this.dataSet.users[userId] = userObject;
-        setImmediate(() => {
-            log.debug("Notifying the listener of RMAU changes");
-            this.onChanges?.({
-                changed: [userId],
-                dataSet: this.dataSet,
-                activeUsers: this.countActiveUsers().allUsers,
-            });
-        });
+        if (!this.debounceTimer) {
+            this.debounceTimer = setTimeout(() => {
+                log.debug("Notifying the listener of RMAU changes");
```

Comment:
```suggestion
                log.debug(`Notifying the listener of RMAU changes for ${Object.keys(this.dataSet).length} users`);
```

Though, I wonder if we should make `users` a Map due to:

- prototype pollution scares
- giving us a `.size` property?

Reply:
That makes sense, yes – though it'll be a breaking API change, since we expose it in the `onChanges`, and IRC and Slack rely on it when storing user activity numbers.

Reply:
Switched it to a proper Map in 141695f – but for the logline change we'd probably want `"changes for ${changed.length} users"` anyway?

Reply:
Actually, this debounce logic breaks it completely, since it always submit a single-element array, so future calls will be ignored if different userIds are used. Time to write some more tests.

Reply:
Fixed now, should be all good.